Datastore/tests/System/FilterTest.php

```diff
@@ -0,0 +1,224 @@
+        }
+        self::$restClient->insertBatch($entities);
+
+        // on rare occasions the queries below are returning no results when
+        // triggered immediately after an insert operation. the sleep here
+        // is intended to help alleviate this issue.
+        sleep(1);
+
+        foreach ($entities as $entity) {
+            self::$localDeletionQueue->add($entity->key());
+        }
+    }
+
+    public static function tear_down_after_class()
+    {
+        self::tearDownFixtures();
+    }
+
+    /**
+     * @dataProvider defaultDbClientProvider
+     */
+    public function testOrRunQuery(DatastoreClient $client)
+    {
+        $this->skipEmulatorTests();
+        $filter = Filter::doOr([
+            Filter::where('Name', '=', 'Hersch'),
+            Filter::where('Name', '=', 'Davy')
+        ]);
+        $query = $client->query()
+            ->kind(self::$kind)
+            ->filter($filter);
+
+        $results = $this->runQueryAndSortResults($client, $query);
+        $this->assertEquals(count($results), 2);
+        $this->assertEquals($results[0]['Name'], 'Hersch');
+        $this->assertEquals($results[1]['Name'], 'Davy');
+    }
+
+    /**
+     * @dataProvider defaultDbClientProvider
+     */
+    public function testMixOfOrAndRunQuery(DatastoreClient $client)
+    {
+        $this->skipEmulatorTests();
+        $filter = Filter::doAnd([
+            Filter::where('Age', '<', 26),
+            Filter::doOr([
+                Filter::doAnd([
+                    Filter::where('Age', '>', 23),
+                    Filter::where('Age', '<', 30)
+                ]),
+                Filter::doAnd([
+                    Filter::where('Age', '>', 25),
+                    Filter::where('Age', '<', 31)
+                ]),
+            ])
+        ]);
+        $query = $client->query()
+            ->kind(self::$kind)
+            ->filter($filter);
+
+        $results = iterator_to_array($client->runQuery($query));
+        $this->assertEquals(count($results), 1);
+        $this->assertEquals($results[0]['Name'], 'Eldredge');
+    }
+
+    /**
+     * @dataProvider defaultDbClientProvider
+     */
+    public function testOrQueryViaTransaction(DatastoreClient $client)
+    {
+        $this->skipEmulatorTests();
+        $filter = Filter::doOr([
+            Filter::where('Name', '=', 'Hersch'),
+            Filter::where('Name', '=', 'Davy')
+        ]);
+        $query = $client->query()
+            ->kind(self::$kind)
+            ->filter($filter);
+
+        $transaction = $client->transaction();
+        $results = iterator_to_array($transaction->runQuery($query));
+        usort($results, function ($a, $b) {
+            return $a['Name'] < $b['Name'];
+        });
+        $this->assertEquals(count($results), 2);
+        $this->assertEquals($results[0]['Name'], 'Hersch');
+        $this->assertEquals($results[1]['Name'], 'Davy');
+        $transaction->commit();
```

Comment:
is this really needed?

Reply:
Assertions are needed right😅. I didn't understand your question I guess. 
Can you please explain more?

Reply:
Yeah since commit would close the transaction. Moreover, that is the suggested way as mentioned [here](https://cloud.google.com/datastore/docs/concepts/transactions#:~:text=In%20production%20code%2C%20it%20is%20important%20to%20ensure%20that%20every%20transaction%20is%20either%20explicitly%20committed%20or%20rolled%20back.).